src/include/iscsi-lio-server/UI_dialogs.rb

```diff
@@ -0,0 +1,2952 @@
+        err_msg = _("Please use username and password in pair.")
+        Yast::Popup.Error(err_msg)
+        return false
+      end
+    end
+    true
+  end
+
+  def store()
+    username = @user_name_input.get_value.gsub(/\s+/,'')
+    password = @password_input.get_value.gsub(/\s+/,'')
+    $discovery_auth.store_userid(username)
+    $discovery_auth.store_password(password)
+  end
+
+  def handle(event)
+    nil
+  end
+
+  def help
+
+  end
+end
+
+class InitiatorAuthDiscovery < CWM::CustomWidget
+  include Yast
+  include Yast::I18n
+  include Yast::UIShortcuts
+  include Yast::Logger
+  def initialize(value)
+    textdomain "iscsi-lio-server"
+    @auth_by_initiator = Auth_by_Initiators_CheckBox.new(self, value)
+    mutual_username = $discovery_auth.fetch_mutual_userid().gsub(/\s+/,'')
+    mutual_password = $discovery_auth.fetch_mutual_password().gsub(/\s+/,'')
+    @mutual_user_name_input = MutualUserName.new(mutual_username)
+    @mutual_password_input = MutualPassword.new(mutual_password)
+    self.handle_all_events = true
+  end
+
+  def contents
+    HBox(
+        HSpacing(25),
+        VBox(
+            Left(@auth_by_initiator,),
+            HBox(
+                @mutual_user_name_input,
+                @mutual_password_input,
+                ),
+            ),
+        HSpacing(25),
+        )
+
+  end
+
+  def disable_checkbox()
+    @auth_by_initiator.set_value(false)
+    @auth_by_initiator.disable()
+  end
+
+  def enable_checkbox()
+    @auth_by_initiator.enable()
+    @auth_by_initiator.value = true
+  end
+
+  def disable_input_fields()
+    @mutual_user_name_input.disable()
+    @mutual_password_input.disable()
+  end
+
+  def enable_input_fields()
+    @mutual_user_name_input.enable()
+    @mutual_password_input.enable()
+  end
+
+  def get_status
+    return @auth_by_initiator.value
+  end
+
+  def opt
+    [:notify]
+  end
+
+  def validate
+    status = @auth_by_initiator.get_value
+    if status == true
+      if (@mutual_user_name_input.get_value == " \n") || (@mutual_password_input.get_value == " \n")
+        err_msg = _("Please use mutual_username and mutual_password in pair.")
+        Yast::Popup.Error(err_msg)
+        return false
+      end
+    end
+    true
+  end
+
+  def store()
+    mutual_username = @mutual_user_name_input.get_value.gsub(/\s+/,'')
+    mutual_password = @mutual_password_input.get_value.gsub(/\s+/,'')
+    $discovery_auth.store_mutual_userid(mutual_username)
+    $discovery_auth.store_mutual_password(mutual_password)
+  end
+
+  def handle(event)
+    nil
+  end
+
+  def help
+    _('')
+  end
+end
+
+class DiscoveryAuthWidget < CWM::CustomWidget
+  include Yast
+  include Yast::I18n
+  include Yast::UIShortcuts
+  include Yast::Logger
+  def initialize()
+    textdomain "iscsi-lio-server"
+    $discovery_auth.analyze()
+    @status = $discovery_auth.fetch_status()
+    if @status == "False \n"
```

Comment:
WHAT??!

Are you seriously representing a boolean value with a string that even includes trailing whitespace? Please fix the parsing method so that `@status` becomes a real boolean.

Reply:
it's because targetcli just return such a string, but I can add a wrap there, thanks for pointing that out!